### DISCUSSION TITLE
Issue restoring files

### DISCUSSION THREAD

**Discussion by btmorr** [bareos/bareos#1991] [Q&A] [answered]
I started [this thread](https://groups.google.com/u/1/g/bareos-users/c/t7mBaL-hNKI) on the mailing list, but as that is going to be going read-only, I'm creating a discussion here to continue the thread.

Issue:
> I am trying to restore files from one of my backups. In the Restore tab of the webui, I select the client and then after a delay I get the following error message:
> 
> {"id":"#","xhr":{"readyState":4,"responseText":"Decoding failed: Syntax errorError: API 2 not available on director.\n                      Please upgrade to verison 15.2.2 or greater and/or compile with jansson support.","status":200,"statusText":"OK"}
> 
> Bareos is running on an Ubuntu 20.04 server with hostname cscisbu01, and I am trying to access a backup from a client with hostname csspfile01. I initially got this error while on `23.0.4~pre169.6644119b5-114`. I upgraded bareos on both the server and client to the most recent version via apt, which brought me up to `23.0.5~pre146.7e91df1c0-130` with no apparent change in behavior.
> 
> I have successfully restored files within the last couple of months on this server from the same client, and have not changed any configurations since then.
> 
> I've attached my apache2 error logs, which contain some possibly-relevant php errors. I didn't see anything relevant in /var/log/bareos/bareos.log or /var/log/bareos/bareos-audit.log, but can provide those or anything else if needed. 

Here is the requested debug trace: [bareos-dir.trace.txt](https://github.com/user-attachments/files/17508491/bareos-dir.trace.txt)

Please let me know if there's any other info needed. Thanks

**Comment by arogge** [comment]
The users mailing-list is *not* going to be read-only.
The devel mailing-list is.

**Comment by btmorr** [comment]
Ah, ok. Apologies for the misread. I tried to reply to my thread, both via email and by the Google Groups interface, and my message did not post, instead showing a line that said "Message has been deleted", so I thought there was an issue with the mailing list. I see that both attempted replies came through some hours later. I'll close this in favor of the email thread

**Comment by joergsteffens** [comment, marked as answer]
Yes, Google Groups did block your message as spam. Don't take this personally, from time to time it is blocking all kinds of messages, including messages from myself. Your email address is now on a whitelist, so that should not happen again.

It is fine to handle your case on bareos-users. Github discussion would also have been okay, but it is good to handle it only in one place.
